src/ripple/core/impl/Config.cpp

```diff
@@ -379,253 +382,253 @@ void Config::load ()
     if (!QUIET)
         std::cerr << "Loading: " << CONFIG_FILE << std::endl;
 
-    std::ifstream   ifsConfig (CONFIG_FILE.c_str (), std::ios::in);
+    std::ifstream ifsConfig (CONFIG_FILE.c_str (), std::ios::in);
 
     if (!ifsConfig)
     {
         std::cerr << "Failed to open '" << CONFIG_FILE << "'." << std::endl;
+        return;
     }
-    else
-    {
-        std::string file_contents;
-        file_contents.assign ((std::istreambuf_iterator<char> (ifsConfig)),
-                              std::istreambuf_iterator<char> ());
-
-        if (ifsConfig.bad ())
-        {
-           std::cerr << "Failed to read '" << CONFIG_FILE << "'." << std::endl;
-        }
-        else
-        {
-            IniFileSections secConfig = parseIniFile (file_contents, true);
-            std::string strTemp;
 
-            build (secConfig);
+    std::string fileContents;
+    fileContents.assign ((std::istreambuf_iterator<char>(ifsConfig)),
+                          std::istreambuf_iterator<char>());
 
-            // XXX Leak
-            IniFileSections::mapped_type* smtTmp;
-
-            smtTmp = getIniFileSection (secConfig, SECTION_VALIDATORS);
-
-            if (smtTmp)
-            {
-                validators  = *smtTmp;
-            }
-
-            smtTmp = getIniFileSection (secConfig, SECTION_CLUSTER_NODES);
+    if (ifsConfig.bad ())
+    {
+        std::cerr << "Failed to read '" << CONFIG_FILE << "'." << std::endl;
+        return;
+    }
 
-            if (smtTmp)
-            {
-                CLUSTER_NODES = *smtTmp;
-            }
+    loadFromString (fileContents);
+}
 
-            smtTmp  = getIniFileSection (secConfig, SECTION_IPS);
+void Config::loadFromString (std::string const& fileContents)
+{
+    IniFileSections secConfig = parseIniFile (fileContents, true);
 
-            if (smtTmp)
-            {
-                IPS = *smtTmp;
-            }
+    build (secConfig);
 
-            smtTmp  = getIniFileSection (secConfig, SECTION_IPS_FIXED);
+    if (auto smtTmp =
```

Comment:
I suggest:

    if (auto s = getIniFileSection (secConfig, SECTION_VALIDATORS))
        validators  = *s;
    
    if (auto s = getIniFileSection (secConfig, SECTION_CLUSTER_NODES))
       CLUSTER_NODES = *s;

    if (auto s = getIniFileSection (secConfig, SECTION_IPS))
        IPS = *s;

    if (auto s = getIniFileSection (secConfig, SECTION_IPS_FIXED))
        IPS_FIXED = *s;
    
    if (auto s = getIniFileSection (secConfig, SECTION_SNTP))
        SNTP_SERVERS = *s;

In particular, `smtTmp` is more cryptic than useful.  :-)